docs/specs/servers/blockchain/blockchain-server-api.md

```diff
@@ -0,0 +1,136 @@
+# Blockchain API Server
+
+## Blockchain RPC
+
+Blockchain RPC API for interacting with different blockchains can be found in a separate `Blockchain RPC` subsection.
+
+## Hexless account profiles API
+
+Email wallet users can register a profile name e.g. `coolname.connect.id` and use it instead of the `0x` address.
+Server API exposes forward lookup for the `name.connect.id` -> `0x` address with chain IDs and reverse lookup for the `0x` -> `name.connect.id`.
+Users can also update the name data including avatar, description text and bind addresses.
+
+### Name lookup
+
+Used to lookup address for the name.
+
+`GET /v1/profile/account/{name}`
+
+* `name` - is the registered name. eg. `coolname.connect.id`
+
+#### Success response body:
+
+```jsonc
+{
+    "name": string,
+    "address": string,
+    "text": (optional)string,
+    "avatar": (optional)string,
+    "registered": datetime,
+    "updated": datetime,
+    "chainIDs" integer[]
+}
+```
+
+* `name` - registered account name.
+* `address` - address that bind to the account name.
+* `text` - account profile self description.
+* `avatar` - URL of the avatar for the account.
```

Comment:
Where are we hosting avatars? Is it better to do base64 encoded data here? Then client can display as a data URI.

Reply:
Hosting images in the Postgres database even as small as 256x256 is a bad idea for the scale and performance that we need. The database size in this case for the 100Mi of users will be monstrous, this is costly, and database pressure (even if using the BLOB type, which is stored out of the table's main index). Zerion, for example, stores images/icons in the S3 storage and just uses links in the API responses.
We definitely should store images and other blobs outside of the DB (s3, storj, ipfs etc.).

Reply:
S3 probably. So probably need the API to accept file uploads and then it will push to S3? Also handling the deletion of the S3 object case if a new avatar is uploaded. What about consistency if upload works but save to Postgres fails?